### ISSUE TITLE
[Non issue] A tool to encode/modify SAN

### ISSUE THREAD

**Issue by Chaython** [clone2727/smushplay#1]
So SMUSH animations are a video codec?
Besides LucasArts' own animation engine, is there a tool to write/modify data in this codec?

**Comment by clone2727** [comment]
All images are a video codec in a way, really :)

I'm unaware of any tools.

**Comment by Chaython** [comment]
Wow, fast reply. I was inquiring, as someone asked about modding Shadows of the Empire in the Steam forum. To which I had basically replied, anything can be modded, there's obviously limitations due to the data being encrypted, but it mostly comes down to a dedicated fan base with talent and time... I was doing some investigation related to modding "Shadows of the Empire". I had assumed since I saw several topics related to this codec/ other software saying they can playback it, I had assumed there must be an application to also create them. https://steamcommunity.com/app/560170/discussions/0/3185654583880485189/
Thanks for your time, friend; and thanks for supporting open source / modding.

**Comment by clone2727** [comment]
Shadows of the Empire is definitely _not_ encrypted. From what I remember (I don't have my CD readily available), most of the game logic is hardcoded into various DLLs. Someone would need to reverse engineer the main binary in order to allow for modding. The textures, etc. are either raw in the data segment or in PE resources (I forget which). And then videos are just loose, along with the WAV sounds and some BMP files for the interface. Interestingly, the BMP files are stored upside-down; I imagine the team working on the game failed to realize that BMP is stored bottom-up.

**Comment by Chaython** [comment]
Yes, I noted the BMP and WAV; obviously they would be easy to swap.. I mentioned that anything encrypted can be decrypted. :P I probably meant hardcoded rather than encrypted...
Perhaps the CD version or something has the BMPs upside down, however in my GOG edition they're right side up, with pink transparency.
All data is stored in .dll
Digital re-releases have an .ogg soundtrack instead of cd-audio as the cd-audio wouldn't playback on modern systems.
and even though GOG usually works with publishers to refresh their games to work on modern hardware along with removing all DRM, the game still relies on a disk dump to launch. The disk data directory is written/reliant to the windows registry for some reason... so if you moved the game to another folder the launcher cannot start the game because it can not load the CD info
..\Star Wars - Shadows of the Empire\INSTALL
There's no information as to where/what the registry keys are; so I have to clean install the game as Windows reassigned my hard drive letter every-time i plugged in a USB drive [so I changed my HDD to B ) :/
[Star Wars - Shadows of the Empire GOG Directory Tree](https://github.com/clone2727/smushplay/files/3650984/Star.Wars.-.Shadows.of.the.Empire.GOG.Directory.zip)
